magenta/lib/sequence_to_melodies.py

```diff
@@ -90,4 +95,57 @@ def basic_one_hot_encoder(melody, steps_per_beat=4, min_note=48, max_note=84,
       'labels': tf.train.FeatureList(feature=label_features)
   }
   feature_lists = tf.train.FeatureLists(feature_list=feature_list)
-  return tf.train.SequenceExample(feature_lists=feature_lists), transpose_amount
\ No newline at end of file
+  return tf.train.SequenceExample(feature_lists=feature_lists), transpose_amount
+
+
+def run_conversion(encoder, sequences_file, train_output, eval_output='', eval_ratio=0.0):
+  """Loop that converts NoteSequence protos to SequenceExample protos.
+
+  Args:
+    encoder: A function that converts Melody to SequenceExample which is fed
+        into a model. The function takes one input of type melodies_lib.Melody
+        and outputs a tuple (tf.train.SequenceExample, reconstruction_data)
+        where reconstruction_data is any extra data that is needed to
+        reconstruct a Melody from the given SequenceExample.
+    sequences_file: String path pointing to TFRecord file of NoteSequence
+        protos.
+    train_output: String path to TFRecord file that training samples will be
+        saved to.
+    eval_output: If set, string path to TFRecord file that evaluation samples
+        will be saved to. Omit this argument to not produce an eval set.
+    eval_ratio: Fraction of input that will be saved to eval set. A random
+        partition is chosen, so the actual train/eval ratio will vary.
+  """
+
+  reader = tf.python_io.tf_record_iterator(sequences_file)
```

Comment:
That's fine. Just wondering though, why have proto-specific wrappers when the TF API supports reading and writing arbitrary serialized protos (and arbitrary bytes)?